tests/core/test_otel_logger.py

```diff
@@ -356,3 +356,110 @@ def test_timer_start_and_stop_manually_send_true(self, mock_time, metrics_consis
         self.meter.get_meter().create_observable_gauge.assert_called_once_with(
             name=full_name(name), callbacks=ANY
         )
+
+    def test_incr_counter(self):
+        # Arrange
```

Comment:
I see your pattern and appreciate that you were taught this way, and may be required in classes, but in the Airflow codebase the Arrange/Act/Assert comments aren't needed.  A blank line separating the sections of the test is sufficient, "everyone" knows what the intent behind the three "sections" is without adding extra lines of text to the code.

I had not seen AAA before, I kinda like that naming.  I was taught it as "Given, When, Then".  Same concept, just different names.